### ISSUE TITLE
Exception of type 'System.OutOfMemoryException' was thrown reading .dcm file

### ISSUE THREAD

**Issue by scapegoat17** [rexcardan/Evil-DICOM#22]
I am trying to upload a .dcm file to read and am getting the exception marked above in the title. My code seems to be stepping into the DICOMObject.Read method without any issues until it gets to this:

(Since i am very new to using GitHub, I would also like to reference my question which is way easier on the eyes to review my code here: https://stackoverflow.com/questions/27954233/can-not-find-open-method-in-evil-dicom-using-latest-version-from-gethub?noredirect=1#comment44335595_27954233)

public static IDICOMElement ReadElementImplicitLittleEndian(DICOMBinaryReader dr)
{
    var tag = TagReader.ReadLittleEndian(dr);
    var vr = TagDictionary.GetVRFromTag(tag);
    int length = LengthReader.ReadLittleEndian(VR.Null, dr);
    var data = DataReader.ReadLittleEndian(length, dr, ransferSyntax.IMPLICIT_VR_LITTLE_ENDIAN);
    var el = ElementFactory.GenerateElement(tag, vr, data,                                                                                                                                  TransferSyntax.IMPLICIT_VR_LITTLE_ENDIAN);
    return el;
}

When the code gets to:

int length = LengthReader.ReadLittleEndian(VR.Null, dr);
length returns an int of 1919252000 bytes which is ~2GB. Then the code steps to:

var data = DataReader.ReadLittleEndian(length, dr, TransferSyntax.IMPLICIT_VR_LITTLE_ENDIAN);
Which checks to see if there are any bytes to read (which there is) and goes to the read bytes here:

public byte[] ReadBytes(int count)
{
    byte[] buffer = new byte[count];
    _binaryReader.Read(buffer, 0, count);
    return buffer;
}
byte[] buffer = new byte[count]; is where the actual exception occurs in the code. I have tested the amount of bytes that it can handle and it seems to be around .6 - .7 GB which is not even half of what I need. Is there away to expand the buffer to accept all of what I need?

**Comment by rexcardan** [comment]
See response on SO.

**Comment by DominaTang** [comment]
I could still meet this "Out of Memory" exception in two cases:
1. When looping throw image folder, if a DICOM file size is very big, say > 500M, DICOMObject.Open() would throw "Out of Memory" exception; If I wrote code just open that particular big size file. everything is good.
2. Sometimes even a DICOM is very small (<1M), it still throws "Out of Memory" exception, I checked Tag values file, might because of some "Group Length" Tag value "FFFFFFFF"? Just a guess.
